packages/manager-components/src/utils/click-utils.ts

```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+
+/**
+ * Assign the function passed to the onClick event handler
+ * as welll as the onKeyDown event handler when using the keys Space or Enter
+ */
+export const handleClick = (
+  fn: React.KeyboardEventHandler & React.MouseEventHandler,
+) => ({
+  onClick: fn,
+  onKeyDown: (event: React.KeyboardEvent) => {
+    if ([' ', 'Enter'].includes(event.key)) {
+      fn(event as React.KeyboardEvent & React.MouseEvent);
```

Comment:
not really fond of casting keyboard event to mouseevent, if you are only interested in "click" event you could define a custom handler type for your "fn" function, maybe something like :

```ts
fn: (ev: Event) => void
```

using a more generic "Event" type make more sense

Reply:
Thank for your feedback